src/main/java/net/snowflake/ingest/connection/SecurityManager.java

```diff
@@ -176,4 +198,82 @@ String getToken()
     return token.get();
   }
 
+  /**
+   * Hashes input bytes using SHA-256.
+   *
+   * @param input  input bytes
+   * @param offset offset into the output buffer to begin storing the digest
+   * @param len    number of bytes within buf allotted for the digest
+   * @return SHA-256 hash
+   */
+  public static byte[] sha256Hash(byte[] input, int offset, int len)
+  {
+    try
+    {
+      MessageDigest md = MessageDigest.getInstance("SHA-256");
+      md.update(input, offset, len);
+      return md.digest();
+    }
+    catch (NoSuchAlgorithmException e)
+    {
+      throw new RuntimeException(e);
+    }
+  }
+  /**
+   * Hashes input bytes using SHA-256.
+   *
+   * @param input input bytes
+   * @return SHA-256 hash
+   */
+  public static byte[] sha256Hash(byte[] input)
+  {
+    return sha256Hash(input, 0, input.length);
+  }
+
+  /**
+   * Hashes input bytes using SHA-256 and converts hash into a string using
+   * Base64 encoding.
+   *
+   * @param input input bytes
+   * @return Base64-encoded SHA-256 hash
+   */
+  public static String sha256HashBase64(byte[] input)
+  {
+    return new String(org.apache.commons.codec.binary.Base64.encodeBase64(sha256Hash(input)));
+  }
+  /**
+   * Given a publickey
+   *
+   * @return the fingerprint of public key
+   */
+  private String getPublicKeyFp(String publicKey)
+  {
+    byte[] rawBytes = org.apache.commons.codec.binary.Base64.decodeBase64(publicKey);
+    return String.format("SHA256:%s", sha256HashBase64(rawBytes));
+  }
+
+
+  /**
+   * Converts PublicKey object to encodedBase64String using RSA scheme.
+   *
+   * @param publicKey
+   * @return
+   * @throws NoSuchAlgorithmException
+   * @throws InvalidKeySpecException
+   */
+  public static String savePublicKey(PublicKey publicKey)
+      throws NoSuchAlgorithmException, InvalidKeySpecException
+  {
+    KeyFactory fact = KeyFactory.getInstance("RSA");
+    X509EncodedKeySpec spec = fact.getKeySpec(publicKey,
+        X509EncodedKeySpec.class);
+    byte[] publicKeyEncode = Base64.getEncoder().encode(spec.getEncoded());
+
+    return new String(publicKeyEncode);
+  }
+
+  public String getPublicKeyFingerPrint()
```

Comment:
The only place you probably need this is tests. So package private should be good enough.